source/postprocess/visualization/dynamic_topography.cc

```diff
@@ -28,19 +28,60 @@ namespace aspect
     namespace VisualizationPostprocessors
     {
       template <int dim>
-      std::pair<std::string, Vector<float> *>
-      DynamicTopography<dim>::execute() const
+      DynamicTopography<dim>::
+      DynamicTopography ()
+        :
+        DataPostprocessorScalar<dim> ("dynamic_topography",
+                                      update_quadrature_points)
+      {}
+
+      template <int dim>
+      void
+      DynamicTopography<dim>::
+      evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                            std::vector<Vector<double> > &computed_quantities) const
       {
+        for (unsigned int q=0; q<computed_quantities.size(); ++q)
+          computed_quantities[q](0) = 0;
+
         const Postprocess::DynamicTopography<dim> &dynamic_topography =
           this->get_postprocess_manager().template get_matching_postprocessor<Postprocess::DynamicTopography<dim> >();
 
-        std::pair<std::string, Vector<float> *>
-        return_value ("dynamic_topography",
-                      new Vector<float>(dynamic_topography.cellwise_topography()));
+        auto cell = input_data.template get_cell<DoFHandler<dim> >();
+
+        bool cell_at_top_or_bottom_boundary = false;
+        for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+          if (cell->at_boundary(f) &&
+              (this->get_geometry_model().translate_id_to_symbol_name (cell->face(f)->boundary_id()) == "top" ||
+               this->get_geometry_model().translate_id_to_symbol_name (cell->face(f)->boundary_id()) == "bottom"))
+            cell_at_top_or_bottom_boundary = true;
+
+        if (cell_at_top_or_bottom_boundary)
+          {
+            std::vector<Point<dim>> quadrature_points(input_data.evaluation_points.size());
```

Comment:
And here something like: "This cell is at the top or bottom, now compute the values of the dynamic topography in this cell." 
It might be just my personal style, but I like very verbose comments. If you feel it clutters the code too much, feel free to leave it out.